plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -315,8 +320,21 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
+        if (type instanceof TimestampType) {
```

Comment:
Please add a test with nested column types, e.g, `row(timestamp)`. I believe the current implementation doesn't work well.

Reply:
Which one do you suggest:
1. Moving those ifs (conversions) to the TypeConverter
2. Reimplement MigrateProcedure to  handle nested types

Reply:
I followed the 2#
I added test for it .